### DISCUSSION TITLE
Auto renewal not working

### DISCUSSION THREAD

**Discussion by kaznad** [polymind-inc/acmebot#403] [Q&A] [answered]
Hi,

Auto renewal is not working for me. I believe that the renewal is done 3 times a week. Monday, Wednesday and Friday at midnight correct?
I have set my local time zone and it did not perform the renewal and even at UTC midnight the renewal was not done. I have a few certs expiring in a couple of weeks. I was able to perform a manual renewal on the dashboard with no issues.

I understand that we can check the error logs in app insight for the function, but I can't find the error logs. Anyone has any idea on this please?

**Comment by shibayan** [comment, marked as answer]
In rare cases, the Timer Trigger of Azure Functions may not work properly, and it will work correctly after running Restart from Azure Portal.

Renewal will only be performed for certificates issued by Acmebot with a validity of less than 30 days.

**Reply by kaznad** [reply]
Hi, you mean to restart the Function right? I will try that.

**Reply by kaznad** [reply]
I restarted the Azure Function, but how do I trigger the renewal? Or do I need to wait till Wednesday to see if it worked?

**Reply by kaznad** [reply]
Hi,

Restarting the Azure Function did not work. The certs are still not getting renewed automatically. Any Ideas?

**Reply by shibayan** [reply]
Make sure that the `RenewCertificates_Timer` function is triggered from Azure Portal.

If triggered, make sure that the Key Vault certificate you expect to be renewed is tagged with `Issuer` and `Endpoint`. Acmebot will only auto-renew certificates that it has issued itself.

**Reply by kaznad** [reply]
Hi, Thank you. I think this might be the issue. I didn't know that the auto-renewal will only work if cert was issued by ACME bot. So after I did an initial manual renewal, the 2 Tags appear in the Key vault. So I believe in 2 months time when it comes for renewal it should work. I will monitor this.

Btw if I may suggest a column in the dashboard to show if the cert is "managed by" acmebot if the 2 tags are present will be useful. So if there tags are not present then it would show not managed by acmebot. Just a suggestion.

**Reply by shibayan** [reply]
Thanks for the feedback. It would be useful to be able to check if it is an update target from the dashboard. I'll try to implement it.

**Reply by shibayan** [reply]
Ref #404

**Reply by ankit-kumar-mck** [reply]
Hey @shibayan ,
For me certificate automatically renewed in 30 days. But i want to renew it 40 days instead of 30 days. Please me know how i should achieve to renew this certificate for 40 days before.

Thanks
